### PR TITLE
domain: migrate test-infra to testify 

### LINKED ISSUE #26306: migrate test-infra to testify for `domain` pkg

### PR DESCRIPTION
**Pull request by tisonkun**
### What problem does this PR solve?

Issue Number: close #26306

This PR is itself ready to review and merge. But it is sadly when run in parallel, there must be some data race so tests become unstable. Also, `TestT` is too complex and couple a lot of thing. It may require further refactor.

I push this PR early to get more comments on the refactor part.

### Release note

- No release note

### CODE REVIEW

File: domain/domain_test.go [hunk 1]
```diff
@@ -40,88 +39,43 @@ import (
...
 	mockStore := &mockEtcdBackend{
 		Storage: s,
-		pdAddrs: []string{clus.Members[0].GRPCAddr()}}
+		pdAddrs: []string{cluster.Members[0].GRPCAddr()}}
+	ddlLease := 80 * time.Millisecond
 	dom := NewDomain(mockStore, ddlLease, 0, 0, mockFactory)
 	defer func() {
 		dom.Close()
-		s.Close()
+		_ = s.Close()
```
Comment (xhebox) [inline domain/domain_test.go:74]: require assertion?

**Review by xhebox — COMMENTED** [review]
Overall LGTM.

One suggestion on parallel tests, you could transform suite into a testing subset manually and parallel multiple subsets. Tests in a suite are not paralleled by default.

File: domain/domain_test.go (hunk 1 above)
Comment (tisonkun) [reply domain/domain_test.go]: Updated.

Let me think of the whole tests.

**Comment by tisonkun** [comment]
@xhebox updated. I think this patch is ready to merge.

ping @tiancaiamao @mmyj @zimulala

**Comment by tisonkun** [comment]
... and yes. I will add an entry about this parallel subset idiom.

**Comment by tiancaiamao** [comment]
This one is so large ... can we split it into smaller PRs?

**Comment by tisonkun** [comment]
@tiancaiamao ok I can break it into three parts.

**Comment by tisonkun** [comment]
@xhebox @tiancaiamao would you take a look at the first part? Or I will push the rest two parts today. Separated parts are for better review and deserve a quick look IMO.

**Pull request closed without merging**